Client/Frontend/Browser/BrowserTrayAnimators.swift

```diff
@@ -103,7 +103,10 @@ private extension BrowserToTrayAnimator {
 
         // Force subview layout on the collection view so we can calculate the correct end frame for the animation
         tabTray.view.layoutSubviews()
-        tabTray.collectionView.scrollToItemAtIndexPath(NSIndexPath(forItem: bvc.tabManager.selectedIndex, inSection: 0), atScrollPosition: .CenteredVertically, animated: false)
+        // don't select anything if the selected index == -1 as that is the non-select value
+        if bvc.tabManager.selectedIndex >= 0 {
```

Comment:
we also set the selected tab to -1 when we delete the current selected tab

On 14 August 2015 at 16:30, Stephan Leroux <notifications@github.com> wrote:

> In Client/Frontend/Browser/BrowserTrayAnimators.swift
> <https://github.com/mozilla/firefox-ios/pull/931#discussion_r37087734>:
>
> > @@ -103,7 +103,10 @@ private extension BrowserToTrayAnimator {
> >
> >          // Force subview layout on the collection view so we can calculate the correct end frame for the animation
> >          tabTray.view.layoutSubviews()
> > -        tabTray.collectionView.scrollToItemAtIndexPath(NSIndexPath(forItem: bvc.tabManager.selectedIndex, inSection: 0), atScrollPosition: .CenteredVertically, animated: false)
> > +        // don't select anything if the selected index == -1 as that is the non-select value
> > +        if bvc.tabManager.selectedIndex >= 0 {
>
> Ah I guess this could happen since we temporarily select -1 when closing
> the last tab right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/firefox-ios/pull/931/files#r37087734>.
>